sdk/src/types/block/payload/signed_transaction/transaction.rs

```diff
@@ -21,86 +22,72 @@ use crate::types::{
+impl TransactionBuilder {
+    /// Creates a new [`TransactionBuilder`].
+    pub fn new(network_id: u64) -> Self {
         Self {
             network_id,
+            creation_slot: None,
             context_inputs: Vec::new(),
             inputs: Vec::new(),
-            inputs_commitment,
-            outputs: Vec::new(),
             allotments: BTreeSet::new(),
             capabilities: Default::default(),
             payload: OptionalPayload::default(),
-            creation_slot: None,
+            outputs: Vec::new(),
         }
     }
 
-    /// Adds creation slot to a [`RegularTransactionEssenceBuilder`].
+    /// Adds creation slot to a [`TransactionBuilder`].
     pub fn with_creation_slot(mut self, creation_slot: impl Into<Option<SlotIndex>>) -> Self {
         self.creation_slot = creation_slot.into();
         self
     }
 
-    /// Adds context inputs to a [`RegularTransactionEssenceBuilder`].
+    /// Adds context inputs to a [`TransactionBuilder`].
     pub fn with_context_inputs(mut self, context_inputs: impl Into<Vec<ContextInput>>) -> Self {
         self.context_inputs = context_inputs.into();
         self
     }
 
-    /// Adds inputs to a [`RegularTransactionEssenceBuilder`].
+    /// Adds inputs to a [`TransactionBuilder`].
     pub fn with_inputs(mut self, inputs: impl Into<Vec<Input>>) -> Self {
         self.inputs = inputs.into();
         self
     }
 
-    /// Adds an input to a [`RegularTransactionEssenceBuilder`].
+    /// Adds an input to a [`TransactionBuilder`].
     pub fn add_input(mut self, input: Input) -> Self {
         self.inputs.push(input);
         self
     }
 
-    /// Adds outputs to a [`RegularTransactionEssenceBuilder`].
-    pub fn with_outputs(mut self, outputs: impl Into<Vec<Output>>) -> Self {
-        self.outputs = outputs.into();
-        self
-    }
-
-    /// Adds an output to a [`RegularTransactionEssenceBuilder`].
-    pub fn add_output(mut self, output: Output) -> Self {
-        self.outputs.push(output);
-        self
-    }
-
-    /// Adds [`ManaAllotment`]s to a [`RegularTransactionEssenceBuilder`].
+    /// Adds [`ManaAllotment`]s to a [`TransactionBuilder`].
     pub fn with_mana_allotments(mut self, allotments: impl IntoIterator<Item = ManaAllotment>) -> Self {
         self.allotments = allotments.into_iter().collect();
         self
     }
 
-    /// Adds a [`ManaAllotment`] to a [`RegularTransactionEssenceBuilder`].
+    /// Adds a [`ManaAllotment`] to a [`TransactionBuilder`].
     pub fn add_mana_allotment(mut self, allotment: ManaAllotment) -> Self {
         self.allotments.insert(allotment);
         self
     }
 
-    /// Replaces a [`ManaAllotment`] of the [`RegularTransactionEssenceBuilder`] with a new one, or adds it.
+    /// Replaces a [`ManaAllotment`] of the [`TransactionBuilder`] with a new one, or adds it.
     pub fn replace_mana_allotment(mut self, allotment: ManaAllotment) -> Self {
```

Comment:
Maybe want to rename this fn in this PR to `set_mana_allotment`, bc there's no noticeable difference to the user between replacing and setting?

Reply:
I wouldn't, at least not in this PR, we have a lot of `replace_` method in the codebase and they would have to be changed for consistency

Reply:
okay